### PR TITLE
fix EIP712 parsing

### PR DESCRIPTION
**Pull request by KillariDev**
Now instead of failing to parse EIP712 message if it doesn't fit to one one ouf identified types, show it as non-identified

Fix https://github.com/DarkFlorist/TheInterceptor/issues/389

### CODE REVIEW

File: app/ts/background/windows/personalSign.ts [hunk 1]
```diff
@@ -112,54 +113,86 @@ export async function craftPersonalSignPopupMessage(ethereumClientService: Ether
...
-
-	if (namedParams.param.primaryType === 'Permit') {
-		const parsed = EIP2612Message.parse(namedParams.param)
-		const token = await getTokenMetadata(ethereumClientService, parsed.domain.verifyingContract)
-		const owner = getAddressMetaData(parsed.message.owner, userAddressBook)
-		if (token.type === 'NFT') throw 'Attempted to perform Permit to an NFT'
+	
+	let parsed 
+	try {
+		parsed = PersonalSignRequestIdentifiedEIP712Message.parse(namedParams.param)
```
[the remark this answers is not in the document]
Comment (KillariDev) [reply app/ts/background/windows/personalSign.ts:119]: yeah, but then I would be catching a lot possible other errors that I do not want to be catching

File: app/ts/background/windows/personalSign.ts [hunk 2]
```diff
@@ -112,54 +113,86 @@ export async function craftPersonalSignPopupMessage(ethereumClientService: Ether
...
-					owner,
-					spender: getAddressMetaData(parsed.message.spender, userAddressBook),
-					verifyingContract: token,
-				},
-				...await getQuarrantineCodes(BigInt(parsed.domain.chainId), account, activeAddressWithMetadata, owner),
+				quarantine: false,
+				quarantineCodes: []
 			}
 		} as const
 	}
```
Comment (MicahZoltu) [inline app/ts/background/windows/personalSign.ts:134]: Don't use try/catch blocks for flow control.  `safeParse` exists for exactly this scenario, where you aren't confident that parsing will succeed and you need to branch on success/failure.  If necessary (doesn't look like it is here), you can access the error details in `maybeParsed.message`, `maybeParsed.fullError`, and `maybeParsed.key` inside of the failure code path.
```suggestion
	const maybeParsed = PersonalSignRequestIdentifiedEIP712Message.safeParse(namedParams.param)
	if (!maybeParsed.success) {
		// if we fail to parse the message, that means it's a message type we do not identify, let's just show it as a nonidentified EIP712 message
		return {
			method: 'popup_personal_sign_request',
			data: {
				originalParams,
				...basicParams,
				type: 'EIP712' as const,
				message: namedParams.param,
				account,
				quarantine: false,
				quarantineCodes: []
			}
		} as const
	}
	const parsed = maybeParsed.value
```

File: app/ts/background/windows/personalSign.ts [hunk 3]
```diff
@@ -112,54 +113,86 @@ export async function craftPersonalSignPopupMessage(ethereumClientService: Ether
...
 	const namedParams = { param: originalParams.params[1], account: originalParams.params[0] }
 	const account = getAddressMetaData(namedParams.account, userAddressBook)
-
-	if (namedParams.param.primaryType === 'Permit') {
-		const parsed = EIP2612Message.parse(namedParams.param)
-		const token = await getTokenMetadata(ethereumClientService, parsed.domain.verifyingContract)
-		const owner = getAddressMetaData(parsed.message.owner, userAddressBook)
-		if (token.type === 'NFT') throw 'Attempted to perform Permit to an NFT'
+
```
Comment (MicahZoltu) [inline app/ts/background/windows/personalSign.ts:116]: Nitpick.  I thought our .editorconfig dealt with this automatically?
```suggestion

```

**Review by MicahZoltu — APPROVED** [review]
Approved, but I *strongly* recommend applying the change I suggested or something similar.  Flow control via exception handling is generally a bad idea.

File: app/ts/background/windows/personalSign.ts (hunk 3 above)
Comment (KillariDev) [reply app/ts/background/windows/personalSign.ts:116]: hmm I think it should

**Pull request merged**